spec/models/course_spec.rb

```diff
@@ -434,4 +434,110 @@
       its(:content_status) { should eq(:published_with_unpublished_changes) }
     end
   end
+
+  describe "#publish_enrichments" do
+    context 'on a course with only a draft enrichment' do
+      let(:course) do
+        create(:course,
+            changed_at: 10.minutes.ago,
+          with_enrichments: [
+          [:initial_draft, created_at: 1.day.ago, updated_at: 20.minutes.ago],
+        ])
+      end
+      let(:user) { create(:user) }
+
+      it 'should publish the draft' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.enrichments.first.status).to eq 'published'
+      end
+
+      it 'should update course changed_at to the current time' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.changed_at).to be_within(1.second).of Time.now.utc
```

Comment:
Same here regarding `reload`. It can be done in the expectation since there's only one expectation here.

Also for the time, you can do:

```
expect(course.reload.changed_at.change(usec: 0)).to eq Time.now.utc.change(usec: 0)
```

Reply:
thanks

* reloads have been tidied up
* I think `to be_within(1.second)` is more expressive